crates/ruff/src/rules/flake8_implicit_str_concat/rules/implicit.rs

```diff
@@ -106,12 +112,95 @@ pub(crate) fn implicit(
                     TextRange::new(a_range.start(), b_range.end()),
                 ));
             } else {
-                diagnostics.push(Diagnostic::new(
+                let mut diagnostic = Diagnostic::new(
                     SingleLineImplicitStringConcatenation,
                     TextRange::new(a_range.start(), b_range.end()),
-                ));
-            }
-        }
+                );
+
+                if let Some(fix) = get_fix_for_single_line_implicit_string_concatenation(
+                    a_tok, *a_range, b_tok, *b_range, locator,
+                ) {
+                    diagnostic.set_fix(fix);
+                }
+
+                diagnostics.push(diagnostic);
+            };
+        };
     }
     diagnostics
 }
+
+fn get_fix_for_single_line_implicit_string_concatenation(
+    a_tok: &Tok,
+    a_range: TextRange,
+    b_tok: &Tok,
+    b_range: TextRange,
+    locator: &Locator,
+) -> Option<Fix> {
+    let (
+        Tok::String {
+            kind: a_kind,
+            triple_quoted: a_triple_quoted,
+            ..
+        },
+        Tok::String {
+            kind: b_kind,
+            triple_quoted: b_triple_quoted,
+            ..
+        },
+    ) = (a_tok, b_tok) else { return Option::None };
+
+    // Fix only strings of the same kind and triple-quotedness
+    if a_kind != b_kind || a_triple_quoted != b_triple_quoted {
+        return Option::None;
+    }
+
+    let a_text = &locator.contents()[a_range];
+    let b_text = &locator.contents()[b_range];
+
+    let a_quotes_style = get_quotes_style(a_text);
+    let b_quotes_style = get_quotes_style(b_text);
+
+    // Fix only strings with the same quotes style
+    if a_quotes_style != b_quotes_style {
+        return Option::None;
+    }
+
+    let text = skip_string_end(*a_triple_quoted, a_text).to_string()
+        + skip_string_start(*b_kind, *b_triple_quoted, b_text);
+
+    Option::Some(Fix::automatic(Edit::range_replacement(
+        text,
+        TextRange::new(a_range.start(), b_range.end()),
+    )))
+}
+
+fn skip_string_start(kind: StringKind, triple_quoted: bool, text: &str) -> &str {
+    let quotes_len = match kind {
+        StringKind::String => 0,
+        StringKind::Bytes | StringKind::FString | StringKind::Unicode | StringKind::RawString => 1,
+        StringKind::RawBytes | StringKind::RawFString => 2,
+    } + (if triple_quoted { 3 } else { 1 });
+
+    &text[quotes_len..]
+}
+
+fn skip_string_end(triple_quoted: bool, text: &str) -> &str {
+    let quotes_len = if triple_quoted { 3 } else { 1 };
+
+    &text[..text.len() - quotes_len]
+}
+
+#[derive(PartialEq)]
+enum QuotesStyle {
+    Double,
+    Single,
+}
+
+fn get_quotes_style(text: &str) -> QuotesStyle {
+    if text.ends_with('\"') {
+        QuotesStyle::Double
+    } else {
+        QuotesStyle::Single
+    }
+}
```

Comment:
The `str` module has some helpers to work with quotes. Can you check if you can re-use some of those?

https://github.com/charliermarsh/ruff/blob/466719247bc68c0b608bc85be341c5aa5e2a5ec2/crates/ruff_python_ast/src/str.rs#L22-L68

Reply:
I don't know why, but these helpers don't work with f-strings.